afsctool.c

```diff
@@ -92,7 +92,7 @@ bool fileIsCompressable(const char *inFile, struct stat *inFileInfo)
 //			ret >= 0 && fsInfo.f_type == 17
 //				&& S_ISREG(inFileInfo->st_mode)
 //				&& (inFileInfo->st_flags & UF_COMPRESSED) == 0 );
-	return (ret >= 0 && (fsInfo.f_type == 17 || fsInfo.f_type == 23 || fsInfo.f_type == 29)
+	return (ret >= 0 && (!strcasecmp("hfs", fsInfo.f_fstypename) || !strcasecmp("apfs", fsInfo.f_fstypename))
 		&& S_ISREG(inFileInfo->st_mode)
```

Comment:
Hmm so would need to add "zfs" here too, since we finally figured out how that decmpfs stuff works (..enough to undo it!).  What WOULD have been neat here is a test like
`VOL_CAPABILITIES_FORMAT] & VOL_CAP_FMT_DECMPFS_COMPRESSION`
So the FS can decide if it supports it or not. Since testing for "hfs" was the reason we had to add mimic-feature :)  rsync and ditto both use capabilites test.

Reply:
Examples of how to check are here
https://opensource.apple.com/source/copyfile/copyfile-146/copyfile.c.auto.html
and here
https://github.com/Homebrew/formula-patches/blob/8f13e94c1f7ae3334b41f60ecd4bb8c406c07941/rsync/patch-hfs-compression.diff#L195-L212